xwayland/xwm.c

```diff
@@ -1246,6 +1247,11 @@ static void xwm_handle_focus_in(struct wlr_xwm *xwm,
 	struct wlr_xwayland_surface *requested_focus = lookup_surface(xwm, ev->event);
 	if (xwm->focus_surface && requested_focus &&
 			requested_focus->pid == xwm->focus_surface->pid) {
+		// We let the application override focus immediately after our focus change
+		// request, but not later
+		if (ev->sequence != xwm->last_focus_seq) {
+			return;
```

Comment:
This ignores focus change events instead of denying them. This means X11 clients can now overwrite the focus (and wlroots' state will be out-of-sync with xserver's).